frontend/src/components/LoginForm.tsx

```diff
@@ -2,34 +2,49 @@ import * as React from 'react';
 
 export type UserHandlerArgs = {
     e: React.FormEvent<HTMLFormElement>;
+    errorMessage: string | null
     username: string | null;
     password: string | null;
 };
 type LoginFormProps = {
     handle_login: ({ e, username, password }: UserHandlerArgs) => void;
+    errorMessage: string | null;
 };
 
-export default function LoginForm({ handle_login }: LoginFormProps) {
+export default function LoginForm({ handle_login, errorMessage }: LoginFormProps) {
     const [username, setUsername] = React.useState<string>(null);
     const [password, setPassword] = React.useState<string>(null);
+    
+    // Clear Error message
+    React.useEffect(() => {
+        errorMessage = ""
+    });
+

```

Comment:
This is problematic. You're passing a prop to a component and immediately overriding it.
If it's part of the component's state, use `useState`. If it's a prop, it's a constant.

Reply:
fixed